File: jambeth/jambeth-merge/src/main/java/de/osthus/ambeth/merge/ICUDResultApplier.java
package de.osthus.ambeth.merge;

import de.osthus.ambeth.cache.ICache;
import de.osthus.ambeth.merge.incremental.IIncrementalMergeState;
import de.osthus.ambeth.merge.model.ICUDResult;

public interface ICUDResultApplier
{
	IIncrementalMergeState acquireNewState(ICache stateCache);

	ICUDResult applyCUDResultOnEntitiesOfCache(ICUDResult cudResult, boolean checkBaseState, final IIncrementalMergeState incrementalState);
}

File: ambeth/Ambeth.Merge/ambeth/merge/ICUDResultApplier.cs
using De.Osthus.Ambeth.Cache;
using De.Osthus.Ambeth.Merge.Incremental;
using De.Osthus.Ambeth.Merge.Model;

namespace De.Osthus.Ambeth.Merge
{
    public interface ICUDResultApplier
    {
        IIncrementalMergeState AcquireNewState(ICache stateCache);

        ICUDResult ApplyCUDResultOnEntitiesOfCache(ICUDResult cudResult, bool checkBaseState, IIncrementalMergeState incrementalState);
    }
}